graph/community/louvain_common.go

```diff
@@ -356,6 +356,10 @@ func (s *slice) TakeMin(p *int) bool {
 const (
 	negativeWeight = "community: unexpected negative edge weight"
 	positiveWeight = "community: unexpected positive edge weight"
+
+	// Threshold for modularity change.
+	// If a modularity gain is lower than this value, we assume there is no gain.
+	modularityEpsilon = 0.0000001
```

Comment:
Please rename this to `deltaQtol`, use e notation and change the comment to 

```
// deltaQtol is the tolerance for progression of the local moving heuristic's improvement of Q.
```

The value seems high compared to your identified problem. Does a value of 1e-14 or thereabouts fix the issue for you?

Reply:
Done, and yes that fixes the issue